tests/influence/test_torch_differentiable.py

```diff
@@ -0,0 +1,177 @@
+"""
+Contains tests for LinearRegression, BinaryLogisticRegression as well as TwiceDifferentiable modules and
+its associated gradient and matrix vector product calculations. Note that there is no test for the neural network
+module.
+"""
+
+import itertools
+from typing import List, Tuple
+
+import numpy as np
+import pytest
+
+from .conftest import (
+    linear_derivative_analytical,
+    linear_hessian_analytical,
+    linear_mixed_second_derivative_analytical,
+    linear_model,
+)
+
+torch = pytest.importorskip("torch")
+import torch
+import torch.nn.functional as F
+from torch import nn
+
+from pydvl.influence.frameworks.torch_differentiable import (
+    TorchTwiceDifferentiable,
+    mvp,
+)
+
+DATA_OUTPUT_NOISE: float = 0.01
+
+TEST_CONDITION_NUMBERS: List[int] = [5]
+TEST_SET_SIZE: List[int] = [20]
+TRAINING_SET_SIZE: List[int] = [50]
+PROBLEM_DIMENSIONS: List[Tuple[int, int]] = [
+    (2, 2),
+    (5, 10),
+    (10, 5),
+    (10, 10),
+]
+
+test_cases_linear_regression_derivatives = list(
```

Comment:
Same comment as above about using `@pytest.mark.parametrize`, and maybe do this in #306